src/lib/time-picker/time-picker-foundation.ts

```diff
@@ -529,7 +531,7 @@ export class TimePickerFoundation implements ITimePickerFoundation {
   }
 
   private _openDropdown(): void {
-    if (!this.allowDropdown) {
+    if (!this.allowDropdown || this._generateTimeOptions.length) {
```

Comment:
_generateTimeOptions is a function, it will always return false here, so isn't accomplishing anything. What could work is instead is changing this to:
const options = this._generateTimeOptions();  // move from line 542
if(!this.allowDropDown || options.length === 0) {
  return;
}

Reply:
actually, i'll revise the above comment, i think it should be:

if (!this.allowDropdown || (!this.showHourOptions && !this.showNow)) {
  return;
}

showHourOptions == false && this.showNow == false are equivalent to not allow drop down as there will be nothing in the drop down to show. If either are true, there should be options in the drop down to show, and worth showing it.

This will remove the later changes in this function below I think.

Reply:
After discussion, we decided on testing options' length. This has been pushed for retesting